compat/mingw.c

```diff
@@ -810,6 +814,15 @@ int mingw_open (const char *filename, int oflags, ...)
 
 	fd = open_fn(wfilename, oflags, mode);
 
+	if (wsl_compat && (oflags & O_CREAT) && fd >= 0) {
+		HANDLE hd = (HANDLE)_get_osfhandle(fd);
+		if (hd != INVALID_HANDLE_VALUE) {
+			struct stat st = { 0 };
+			st.st_mode = S_IFREG | (mode & 0777);
+			write_wsl_stat(hd, &st);
```

Comment:
While this is correct, it would be more obvious what `write_wsl_stat()` does if we explicitly passed uid/gid/mode parameters to it instead of a full (and faked) `struct stat` instance.

Also, how about renaming it to `copy_wsl_mode_bits_to_disk()`?

Further, there is only one caller of this function. It would probably make more sense to feed `int fd` to it, for readability, and do the `_get_osfhandle()` dance inside that function.

Lastly, I wonder whether it would make the feature even better encapsulated if the `wsl_compat` check happened inside that function, too?